service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformUtils.java

```diff
@@ -62,6 +64,53 @@ public static void setCommandError(Command command, Exception e) {
                 .build());
     }
 
+    /**
+     * Delete the Terraform state directory for the given request under the given directory.<br>
+     * The directory name is constructed as follows: {context}-{serviceId}
+     *
+     * @param request
+     * @param directory
+     */
+    public static void deleteConfigPath(CommandRequest request, String directory) {
+        // just in case the delete operation is called concurrently
+        synchronized (lock) {
+            Validate.notNull(request, "request must be provided");
+            Validate.notEmpty(request.getContext(), "context of request must be provided");
+            Validate.notEmpty(request.getServiceId(), "serviceId of request must be provided");
+            Validate.notEmpty(directory, "directory must be provided");
```

Comment:
Validations can be outside synchronized block. Its in our interest to keep the lock only to the critical section of the code